python/ray/rllib/agents/agent.py

```diff
@@ -2,797 +2,7 @@
-            sample_async=config["sample_async"],
-            compress_observations=config["compress_observations"],
-            num_envs=config["num_envs_per_worker"],
-            observation_filter=config["observation_filter"],
-            clip_rewards=config["clip_rewards"],
-            clip_actions=config["clip_actions"],
-            env_config=config["env_config"],
-            model_config=config["model"],
-            policy_config=config,
-            worker_index=worker_index,
-            monitor_path=self.logdir if config["monitor"] else None,
-            log_dir=self.logdir,
-            log_level=config["log_level"],
-            callbacks=config["callbacks"],
-            input_creator=input_creator,
-            input_evaluation=input_evaluation,
-            output_creator=output_creator,
-            remote_worker_envs=config["remote_worker_envs"],
-            remote_env_batch_wait_ms=config["remote_env_batch_wait_ms"],
-            soft_horizon=config["soft_horizon"],
-            _fake_sampler=config.get("_fake_sampler", False))
-
-    @override(Trainable)
-    def _export_model(self, export_formats, export_dir):
-        ExportFormat.validate(export_formats)
-        exported = {}
-        if ExportFormat.CHECKPOINT in export_formats:
-            path = os.path.join(export_dir, ExportFormat.CHECKPOINT)
-            self.export_policy_checkpoint(path)
-            exported[ExportFormat.CHECKPOINT] = path
-        if ExportFormat.MODEL in export_formats:
-            path = os.path.join(export_dir, ExportFormat.MODEL)
-            self.export_policy_model(path)
-            exported[ExportFormat.MODEL] = path
-        return exported
-
-    def __getstate__(self):
-        state = {}
-        if hasattr(self, "local_evaluator"):
-            state["evaluator"] = self.local_evaluator.save()
-        if hasattr(self, "optimizer") and hasattr(self.optimizer, "save"):
-            state["optimizer"] = self.optimizer.save()
-        return state
-
-    def __setstate__(self, state):
-        if "evaluator" in state:
-            self.local_evaluator.restore(state["evaluator"])
-            remote_state = ray.put(state["evaluator"])
-            for r in self.remote_evaluators:
-                r.restore.remote(remote_state)
-        if "optimizer" in state:
-            self.optimizer.restore(state["optimizer"])
-
-    def _register_if_needed(self, env_object):
-        if isinstance(env_object, six.string_types):
-            return env_object
-        elif isinstance(env_object, type):
-            name = env_object.__name__
-            register_env(name, lambda config: env_object(config))
-            return name
-        raise ValueError(
-            "{} is an invalid env specification. ".format(env_object) +
-            "You can specify a custom env as either a class "
-            "(e.g., YourEnvCls) or a registered env id (e.g., \"your_env\").")
+# TODO(ekl) remove legacy agent name
+Agent = Trainer
```

Comment:
Maybe add some deprecation warning here if Agent is used?

Reply:
Done